File: solutions/go/JumpGameII.go
func jump(nums []int) int {
	var step, cur, maxReach int

	for i := 0; i < len(nums)-1; i++ {
		if tmp := nums[i] + i; tmp > maxReach {
			maxReach = tmp
		}

		if cur == i {
			step++
			cur = maxReach
		}
	}

	return step
}

File: solutions/cpp/JumpGameII.cc
class Solution {
public:
    int jump(vector<int>& nums) {
        int step = 0, maxReach = 0, cur = 0;

        for (int i = 0; i < nums.size()-1; ++i) {
            maxReach = std::max(maxReach, nums[i]+i);
            if (i == cur) {
                ++step;
                cur = maxReach;
            }
        }

        return step;
    }
};

